File: 2015/day_1/main.rs
use std::fs::File;
use std::io::prelude::*;

fn main() -> std::io::Result<()> {
    let mut floor = 0;
    let mut pos = 0;
    let mut part2 = 0;
    let mut file = File::open("input.txt")?;
    let mut contents = String::new();
    file.read_to_string(&mut contents)?;
    
    for c in contents.chars() {
        pos += 1;
        if c == '(' {
            floor += 1;
        } else if c == ')' {
            floor -= 1;
        }

        if floor == -1  && part2 == 0 {
            part2 = pos;
        }
    }

    println!("Part 1 answer: {}, Part 2 answer: {}", floor, part2);
    Ok(())
}

File: 2015/day_1/main.py
def main(fn: str) -> None:
    floor: int = 0
    with open(fn, encoding='utf-8') as file:
        while 1:
            char: str = file.read(1)
            if not char:
                break
            print(char)
            if char == '(':
                floor += 1
            elif char == ')':
                floor -= 1
        print("Santa is on floor: " + str(floor))

main("input.txt")
